File: php/src/CodingInterview/Domain/Stock.php
<?php

declare(strict_types=1);

namespace Folio\CodingInterview\Domain;

/** 保有銘柄（銘柄と保有額）を表す。 */
final class Stock
{
    public function __construct(
        public readonly StockSymbol $symbol,
        public readonly BigDecimal $amountJpy,
    ) {
    }
}

File: java8/src/main/java/folio/codinginterview/domain/Stock.java
package folio.codinginterview.domain;

import java.math.BigDecimal;

// 銘柄と保有額を表す。
public final class Stock {
    private final StockSymbol symbol;
    private final BigDecimal amountJpy;

    public Stock(StockSymbol symbol, BigDecimal amountJpy) {
        this.symbol = symbol;
        this.amountJpy = amountJpy;
    }

    public StockSymbol symbol() { return symbol; }

    public BigDecimal amountJpy() { return amountJpy; }
}
